solidity/contracts/bridge/Bridge.sol

```diff
@@ -408,6 +418,59 @@ contract Bridge is Ownable, EcdsaWalletOwner {
         self.notifyRedemptionTimeout(walletPubKeyHash, redeemerOutputScript);
     }
 
+    /// @notice Submits the moving funds target wallets commitment.
+    ///         Once all requirements are met, that function registers the
+    ///         target wallets commitment and opens the way for moving funds
+    ///         proof submission.
+    /// @param walletPubKeyHash 20-byte public key hash of the source wallet
+    /// @param walletMainUtxo Data of the source wallet's main UTXO, as
+    ///        currently known on the Ethereum chain.
+    /// @param walletMembersIDs Identifiers of the source wallet signing group
+    ///        members
+    /// @param walletMemberIndex Position of the caller in the source wallet
+    ///        signing group members list
+    /// @param targetWallets List of 20-byte public key hashes of the target
+    ///        wallets that the source wallet commits to
+    /// @dev Requirements:
+    ///      - The source wallet must be in the MovingFunds state
+    ///      - The source wallet must not have pending redemption requests
+    ///      - The source wallet must not have prior commitments submitted earlier
+    ///      - The expression `keccak256(abi.encode(walletMembersIDs))` must
+    ///        be exactly the same as the hash stored under `membersIdsHash`
+    ///        for the given source wallet in the ECDSA registry. Those IDs are
+    ///        not directly stored in the contract for gas efficiency purposes
+    ///        but they can be read from appropriate `DkgResultSubmitted`
+    ///        and `DkgResultApproved` events.
+    ///      - The `walletMemberIndex` must be in range [1, walletMembersIDs.length]
+    ///      - The caller must be the member of the source wallet signing group
+    ///        at the position indicated by `walletMemberIndex` parameter
+    ///      - The `walletMainUtxo` components must point to the recent main
+    ///        UTXO of the source wallet, as currently known on the Ethereum
+    ///        chain. If the source wallet has no main UTXO, this parameter
+    ///        can be empty as it is ignored since the wallet balance is
+    ///        assumed to be zero.
```

Comment:
Does this part make sense for this function? If there is no main UTXO, there is no moving funds, right?

Reply:
Technically, this is possible because a wallet can have 0 BTC and be moved to the `MovingFunds` state. Of course, we'll handle that case with the "low balance close" function and a rational wallet should call it instead of submitting a commitment.

Reply:
Shouldn't we instead revert if there is no main UTXO? As you say, in this case, calling this function does not make sense and the system is entering into a weird state where one wallet member can declare to which live wallets it is moving the zero balance. I would rather not allow a wallet to enter the path it should not be on.

Reply:
Also, please see this discussion: https://github.com/keep-network/tbtc-v2/pull/213#discussion_r856944271.

Reply:
I did not go with reverting here because before we agreed that `movingFunds` function should no longer transition to `Closed/Closing` state (https://github.com/keep-network/tbtc-v2/pull/213#discussion_r856944271) and we will have a function to close wallets with a balance below the dust threshold as a shortcut. The "no main UTXO set" case is just a specific case of a balance below the dust threshold. If we want to keep `movingFunds` function closing the wallets below the dust immediately and revert upon commitment in the same case without having a separate function for low balance close, we can run into troubles. Suppose such a case:
- Wallet has 1001 satoshi and the dust threshold is 1000 sat
- Wallet is requested to move its funds and the `balance < dust` condition is false so the wallet should push a commitment and cannot be immediately closed
- Meanwhile, the governance change the dust threshold to be 1500 sat 
- Wallet cannot submit the commitment because `balance < dust` is true now and gets slashed eventually.

Reply:
What I am trying to say in https://github.com/keep-network/tbtc-v2/pull/213#discussion_r856944271 is that if the wallet is closeable and has just the dust (or nothing) the `notifyCloseable` function should not put it into MovingFunds state but put it into Closing state immediately.

If we do it this way, we are not asking a wallet that has no funds (or just a dust of funds) to move funds and hope that wallet will not commit to a target wallets list. 

If the wallet has no funds (or just dust) we should not ask it to move funds at all and do not even give it a chance to commit to target wallets because MovingFunds state for no funds or dust of funds does not make sense.

Regarding the scenario you mentioned, we do not need to revert from this function if the balance is below the dust but I *think* we should revert if there is no main UTXO because first, the balance is zero so there is nothing to move, and second, proving the move of funds with no main UTXO is not possible. In practice, for no mainUTXO case, there should be no scenario when this function is callable because closeable notification should turn the wallet into Closing state immediately (?)

Reply:
So let's summarize. You propose:
- `moveFunds` should allow putting the wallet into `Closing` state. We just change that this occurs in the case when the balance is below dust, not just when main UTXO is not set
- `submitMovingFundsCommitment` reverts if main UTXO is not set (balance is zero) but doesn't care about the dust threshold
- There is a separate function for low balance close because there can be a case where `moveFunds` does not close the wallet but after handling all pending redemptions, the wallet can have a balance below dust and may not be able to transfer such a low balance to another wallet

The previous version (which I tried to implement here and in #139) was:
- `moveFunds` always put the wallet into `MovingFunds` without taking care of the balance
- `submitMovingFundsCommitment` allows putting a commitment as soon as there are no pending redemptions and doesn't care about the balance
-  There is a separate function for low balance close that can be called if the wallet is in `MovingFunds` and the balance is below dust. It can be called even if the commitment was submitted

Your proposition makes sense but makes the code and wallet lifecycle diagram a little bit more complex. Are we sure that this is justified well?

Reply:
Ok, let's do it. Will push changes around the commitment in this PR and those about `moveFunds` in #139. The additional function for low balance close will be a new PR anyway.

Reply:
See c94ffd57